### PR TITLE
Decouple action buttons from the editor. Expose action functions

### LINKED ISSUE #39: Move actions *outside* of builder
* In order to be easier to use as a component inside an existing application, it might make sense to remove the action buttons and provide function calls to achieve the same.
* Move the buttons to the builder wrapper used on the build.gobl.org site.

### PR DESCRIPTION
**Pull request by beliolfa**
- Add optional prop to `hideActionButtons`
- Expose functions to perform the same actions that Action Buttons do from outside
- Refactor validEditor computed to be part of the store so it can be reused
- Expose two-way bindings to know the state of `isSigned` and `isValid`

Closes #39

### CODE REVIEW

File: src/lib/GOBLBuilder.svelte [hunk 1]
```diff
@@ -27,12 +36,24 @@
   // generated and used for signing GOBL documents.
   export let signEnabled = true;
 
+  // When enabled, actions buttons such as Build, Sign and Validate are hidden. The underlying actions
+  // that they fire can be still executed from outside.
+  export let hideActionButtons = false;
+
+  // Menu instance to be able to call functions from outside
+  let menu: MenuBar;
```
Comment (samlown) [inline src/lib/GOBLBuilder.svelte:44]: It does feel strange having a menu bar with hidden actions... would it make sense to have the actions defined somewhere explicitly without them being UI components?

**Review by samlown — COMMENTED** [review]
Looking good from the outset! But I can't help thinking this is a good opportunity to refactor the Builder and de-couple the actions from the UI buttons.

File: src/lib/GOBLBuilder.svelte [hunk 2]
```diff
@@ -27,12 +36,24 @@
   // generated and used for signing GOBL documents.
   export let signEnabled = true;
 
+  // When enabled, actions buttons such as Build, Sign and Validate are hidden. The underlying actions
+  // that they fire can be still executed from outside.
+  export let hideActionButtons = false;
```
[the remark this answers is not in the document]
Comment (beliolfa) [reply src/lib/GOBLBuilder.svelte:41]: Let's go all in then! I will remove completely the buttons.

**Review by samlown — APPROVED** [review]
Looking great! I've added some suggestions to consider as well.

File: src/lib/editor/Editor.svelte [hunk 3]
```diff
@@ -195,8 +196,15 @@
...
-    envelope.subscribe((value) => {
-      monacoEditor.updateOptions({ readOnly: Boolean(value?.sigs) });
+    envelope.subscribe(async (value) => {
+      const isSigned = Boolean(value?.sigs);
+      if (!isSigned) {
+        monacoEditor.updateOptions({ readOnly: false });
+        return;
+      }
+
+      await tick();
```
Comment (samlown) [inline src/lib/editor/Editor.svelte:206]: Nice! Given that it's not obvious what `tick()` means, I'd strongly suggest adding some comments around this to explain the situation!

Comment (cavalle) [reply src/lib/editor/Editor.svelte:206]: Agreed!

**Pull request merged**